src/main/kotlin/net/ccbluex/liquidbounce/utils/block/placer/SupportFeature.kt

```diff
@@ -0,0 +1,140 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LiquidBounce. If not, see <https://www.gnu.org/licenses/>.
+ */
+package net.ccbluex.liquidbounce.utils.block.placer
+
+import net.ccbluex.liquidbounce.config.ToggleableConfigurable
+import net.ccbluex.liquidbounce.utils.block.getState
+import net.ccbluex.liquidbounce.utils.block.isBlockedByEntities
+import net.ccbluex.liquidbounce.utils.block.manhattanDistanceTo
+import net.ccbluex.liquidbounce.utils.client.Chronometer
+import net.ccbluex.liquidbounce.utils.collection.Filter
+import net.ccbluex.liquidbounce.utils.math.sq
+import net.minecraft.block.Blocks
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.math.Direction
+import java.util.*
+import kotlin.Comparator
+
+// TODO support no wall range, proper reach calculations
+// wall range support could be done by taking the angle and, if the face is pointing to the player, exclude
+// TODO cache blocked / allowed spots
+/**
+ * Places blocks to support placements.
+ */
+class SupportFeature(val placer: BlockPlacer) : ToggleableConfigurable(placer, "Support", true) {
+
+    private val depth by int("Depth", 4, 1..12)
+    val delay by int("Delay", 500, 0..1000)
+
+    // what block helping blocks can be, by default just "trash blocks", meaning very common blocks
+    val filter by enumChoice("Filter", Filter.WHITELIST)
+    val blocks by blocks("Blocks", hashSetOf(Blocks.COBBLESTONE, Blocks.DIRT, Blocks.NETHERRACK))
+
+    // we don't have to consistently search, every once in a while is okay, see `delay`
+    val chronometer = Chronometer()
+
+    // positions we may not place at
+    val blockedPositions = hashSetOf<BlockPos>()
+
+    /**
+     * Currently finds the best path of blocks to support the placement of [targetPos] using Dijkstra's algorithm,
+     * the speed can possibly be improved by adding heuristics and making it an A* algorithm.
+     */
+    fun findSupport(targetPos: BlockPos): Set<BlockPos>? {
+        val rangeSq = placer.range.sq()
+
+        val openList = PriorityQueue<Node>(Comparator.comparingDouble { it.totalCost })
+        val closedList = hashSetOf<BlockPos>()
+
+        val startNode = Node(targetPos, null, 0.0)
+        openList.add(startNode)
+
+        while (!openList.isEmpty()) {
+            val currentNode = openList.poll()
+            closedList.add(currentNode.position)
+
+            // found a possible path
+            if (canPlace(currentNode.position)) {
+                return reconstructPath(currentNode)
+            }
+
+            for (direction in Direction.entries) {
+                val neighbor = currentNode.position.offset(direction)
+
+                // skip visited nodes
+                if (closedList.contains(neighbor)) {
+                    continue
+                }
```

Comment:
The function is nested too deeply and too complex, detekt even told you about it, but you just adjusted the baseline lol. Anyway, you can just extract everything after this if statement into it's own function like `exploreNeighbor(neighbor)` and the problem is solved.